### ISSUE TITLE
Crash when repairing corrupted world with backup

### ISSUE THREAD

**Issue by p0xus** [Fenixin/Minecraft-Region-Fixer#44]
Bug report:

<type 'exceptions.IndexError'>
Traceback (most recent call last):
  File "C:...\Minecraft-Region-Fixer-master\regionfixer.py
", line 506, in <module>
    value = main()
  File "C:...\Minecraft-Region-Fixer-master\regionfixer.py
", line 433, in main
    fixed = w.replace_problematic_chunks(backup_worlds, problem, ent_lim, del_en
t)
  File "C:...\Minecraft-Region-Fixer-master\regionfixer_co
re\world.py", line 906, in replace_problematic_chunks
    coords = get_region_coords(backup_region_path.split()[1])
IndexError: list index out of range

**Comment by mellamokb** [comment]
I am getting this same problem, trying to replace 8 wrong-located chunks from a backup world.

type 'exceptions.IndexError'
Traceback (most recent call last):
  File "regionfixer.py", line 506, in <module>
  File "regionfixer.py", line 433, in main
  File "regionfixer_core\world.pyo", line 906, in replace_problematic_chunks
  File "regionfixer_core\world.pyo", line 1150, in get_region_coords
IndexError: list index out of range

**Comment by Fenixin** [comment]
Hello!

Thanks for the bug report. This is a more or less known bug. Will try to fix it in the next patch!

**Comment by Maxetto** [comment]
Hope this fix comes soon as I have the same error and a corrupted Chunk which crashes my server.

**Comment by Fenixin** [comment]
@Maxetto, @mellamokb, @Tony-Yates: Could you send me a world/region file to reproduce this problem? I have an idea of what is going on with the tracebacks posted here but I'm unable to reproduce the problem.

**Comment by Maxetto** [comment]
@Fenixin Here it is: https://mega.nz/#!AhZ2zJgR!tOH1a-X9D5nk0rQV7pWkEvjn8HjOF9fOPcJNX-eXbKg

**Comment by Fenixin** [comment]
@Maxetto: I think I have found the problem. If it's what it seems, is a very easy thing to fix and you can test it yourself. Does your backups world path have spaces or strange characters in it? If that is the case write the backups path using quotations marks, that workarounded the problem for me. For example, in the command line, in the options --backups, instead of:
```
... --backups ~/minecraft worlds/bug-test-backups ...
```
Write:
```
... --backups "~/minecraft worlds/bug-test-backups" ...
```

I hope that works. Please, report back with your findings!

I'll try to make a new version with a fix for this as soon as possible.

**Comment by Fenixin** [comment]
@Maxetto: Forget what I said, I have found the problem and needs a fix or it won't replace chunks. I have uploaded a branch called "2to3" (is a WIP). Please, if you don't know how to run the code read:

https://github.com/Fenixin/Minecraft-Region-Fixer/wiki/Installation

Don't hesitate to ask if you need any help.

**Comment by Postremus** [comment]
@Fenixin I had the same problem yesterday. works now with d4b7af656c4e8e94c8248a49ca1895a7132c792d
thanks for the great work!

**Comment by Fenixin** [comment]
This has been fixed thanks to the information provided in #61. Closing it!
